prisma/seed/seed.ts

```diff
@@ -0,0 +1,93 @@
+import { createSeedClient } from '@snaplet/seed';
+import { faker } from '@faker-js/faker';
+
+const main = async () => {
+  const seed = await createSeedClient({
+    connect: true,
+  });
+
+  await seed.$resetDatabase();
+
+  await seed.user((createMany) =>
+    createMany(5, () => ({
+      authId: faker.string.uuid(),
+      email: faker.internet.email(),
+      firstName: faker.person.firstName(),
+      lastName: faker.person.lastName(),
+      role: faker.helpers.arrayElement(['TEACHER', 'ADMIN']),
+      status: faker.helpers.arrayElement(['ACTIVE', 'INVITED', 'DEACTIVATED']),
+    }))
+  );
+
+  const generateSchoolNameAbbreviation = (schoolName: string): string => {
+    const words = schoolName.split(' ');
+    const abbreviation = words
+      .map((word) => word.charAt(0).toUpperCase())
+      .join('');
+    return abbreviation;
+  };
+
+  await seed.location((createMany) =>
+    createMany(5, () => {
+      const school = faker.helpers.arrayElement([
+        'Lambton Park Community School',
+        'St Martin de Porres Catholic School',
+        'Yorkwoods Public School',
+        'Parkdale Junior Senior Public School',
+      ]);
+      return {
+        name: school,
+        abbreviation: generateSchoolNameAbbreviation(school),
+      };
+    })
+  );
+
+  const generateSubjectAbbreviation = (subjectName: string): string => {
+    const words = subjectName.split('_');
+    const abbreviation = words
+      .map((word) => word.charAt(0).toUpperCase())
+      .join('');
+    return abbreviation;
+  };
+
+  await seed.subject((createMany) =>
+    createMany(5, () => {
+      const subject = faker.helpers.arrayElement([
+        'STRINGS',
+        'M_AND_M',
+        'CHOIR',
+        'PERCUSSION',
+      ]);
+      return {
+        name: subject,
+        abbreviation: generateSubjectAbbreviation(subject),
+      };
+    })
+  );
+
+  await seed.absence((createMany) =>
+    createMany(10, () => ({
```

Comment:
doesn't this need to have a absentTeacher, subject, and location? It should have all the required fields here: https://www.notion.so/uwblueprintexecs/Data-Models-d5cdb66530834c8da4ace4184bb363f2#dd896001196c4e14a6e248ab7510af1f

Reply:
Resolved. Should be working. We bing chilling.